app/github_repo_fetcher.rb

```diff
@@ -63,6 +50,27 @@ def latest_commit(repo_name, path)
     }
   end
 
+  def recursively_fetch_files(repo_name, path)
+    docs = client.contents("alphagov/#{repo_name}", path: path)
+    top_level_files = docs.select { |doc| doc.name.end_with?(".md") }.map do |doc|
+      contents = HTTP.get(doc.download_url)
+      filename = doc.path.sub("docs/", "").match(/(.+)\..+$/)[1]
+      title = ExternalDoc.title(contents) || filename
+      {
+        path: "/apps/#{repo_name}/#{filename}.html",
+        title: title.to_s.force_encoding("UTF-8"),
+        markdown: contents.to_s.force_encoding("UTF-8"),
+        relative_path: doc.path,
+        source_url: doc.html_url,
+        latest_commit: latest_commit(repo_name, doc.path),
+      }
+    end
+    nested_files = docs.select { |doc| doc.type == "dir" }.reduce([]) do |memo, dir|
+      memo.concat(recursively_fetch_files(repo_name, dir.path))
+    end
+    top_level_files.concat(nested_files)
```

Comment:
I've got an alternative suggestion to avoid the `nested_files` variable and using `each_with_object` (just a personal preference think, find it easier to read than `reduce`). 

```rb
    docs.select { |doc| doc.type == "dir" }.each_with_object(top_level_files) do |dir, files|
      files += recursively_fetch_files(repo_name, dir.path)
    end
```

Reply:
Nice 👍  (the `+=` doesn't work but the `concat` does)